public/app/features/dashboard-scene/sharing/ExportButton/ExportAsImage.tsx

```diff
@@ -0,0 +1,263 @@
+  useEffect(() => {
+    return () => {
+      if (imageBlob) {
+        URL.revokeObjectURL(URL.createObjectURL(imageBlob));
+      }
+    };
+  }, [imageBlob]);
+
+  const onExport = async () => {
+    setIsLoading(true);
+    setError(null);
+
+    try {
+      const result = await generateDashboardImage({
+        dashboard,
+        scale: config.rendererDefaultImageScale || 1,
+      });
+
+      if (result.error) {
+        throw new Error(result.error);
+      }
+
+      setImageBlob(result.blob);
+      DashboardInteractions.generateDashboardImageClicked({
+        scale: config.rendererDefaultImageScale || 1,
+        shareResource: 'dashboard',
+        success: true,
+      });
+    } catch (error) {
+      console.error('Error exporting image:', error);
+      DashboardInteractions.generateDashboardImageClicked({
+        scale: config.rendererDefaultImageScale || 1,
+        shareResource: 'dashboard',
+        success: false,
+        error: error instanceof Error ? error.message : 'Failed to generate image',
+      });
+      setError({
+        title: t('share-modal.image.error-title', 'Failed to generate image'),
+        message: error instanceof Error ? error.message : 'Failed to generate image',
+      });
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const onDownload = () => {
+    if (!imageBlob) {
+      return;
+    }
+
+    const time = new Date().getTime();
+    const name = dashboard.state.title;
+    saveAs(imageBlob, `${name}-${time}.png`);
+
+    DashboardInteractions.downloadDashboardImageClicked({
+      fileName: `${name}-${time}.png`,
+      shareResource: 'dashboard',
+    });
+  };
+
+  if (!config.rendererAvailable) {
+    return <RendererAlert />;
+  }
+
+  return (
+    <>
+      <p className={styles.info}>
+        <Trans i18nKey="share-modal.image.info-text">Save this dashboard as an image.</Trans>
+      </p>
+
+      <div className={styles.buttonRow}>
+        {!imageBlob ? (
+          <Button
+            variant="primary"
+            onClick={onExport}
+            disabled={isLoading}
+            icon="document-info"
+            data-testid={selectors.components.ExportImage.buttons.generate}
+          >
+            <Trans i18nKey="share-modal.image.generate-button">Generate image</Trans>
+          </Button>
+        ) : (
+          <Button
+            variant="primary"
+            onClick={onDownload}
+            icon="download-alt"
+            data-testid={selectors.components.ExportImage.buttons.download}
+          >
+            <Trans i18nKey="share-modal.image.download-button">Download image</Trans>
+          </Button>
+        )}
+        <Button
+          variant="secondary"
+          onClick={onDismiss}
+          fill="outline"
+          data-testid={selectors.components.ExportImage.buttons.cancel}
+        >
+          <Trans i18nKey="share-modal.image.cancel-button">Cancel</Trans>
+        </Button>
+      </div>
+
+      <div
```

Comment:
This is always shown whereas in the panel > share link view, we see it only after clicking on "generate image" and there is also a "panel title" text. I think it could be nice to actually share the same component between these two to ensure users always have the same preview experience.